### DISCUSSION TITLE
Blog: 2024 Protocol Roadmap

### DISCUSSION THREAD

**Discussion by bnewbold** [bluesky-social/atproto#2469] [General Discussion]
This is a discussion thread for the [2024 Protocol Roadmap blog post](https://docs.bsky.app/blog/2024-protocol-roadmap).

**Comment by MasterJ93** [comment]
Looking good! I'm mostly excited for E2EE and OAuth: interested in seeing how that will work in practice.

**Comment by emilyhunt** [comment]
> Topic/Community Feeds: one of the more common uses for feed generators is to categorize content by topic or community. These feeds are not personalized (they look the same to all users), are not particularly "algorithmic" (posts are either in the feed or not), and often have relatively clear inclusion criteria (though they may be additionally curated or filtered). We are exploring ways to make it easier to create, curate, and explore this type of feed.

It would be great to have easier ways to run topical/curated feeds! I have some thoughts from running [the astronomy feed](https://bsky.app/profile/did:plc:jcoy7v3a2t4rcfdh6i4kza25/feed/astro) over the past year, which is a feed with a curated list of members. I think that curation is vital to the success of it and other science feeds: preventing disinformation and ensuring high post quality is really important to us, and having a curated poster list really helps with that.

The most helpful feature I can think of would be some in-app button that we could put on feeds that people can use to apply to join the poster list. I think it would be an "apply" button, which then opens a pop-up asking them to accept feed rules, after which the application is sent. The protocol part is that this application would have to be sent to the feed server somehow, _and/or_ distributed to a list of assigned feed moderators in-app to accept/reject. This would be a _huge_ help to the current messy workflows of feeds like the Science and Astronomy feeds when we add people to the feed.

An alternative way to run this that could be easier and uses existing parts of the protocol would be to enhance lists with more functionality. The Science feed uses in-app lists of users for moderation (whereas the Astronomy one uses a list on our database linked to an off-site signup form). Maybe you could make it so that lists can have an "apply" button and multiple moderators? That would enhance two features at once (both feeds _and_ lists). Eventually, maybe some clearer way of linking the feed and its associated list together would be good (e.g. in the "about" page of the feed, or with a dedicated tab on the feed overview). @Bossett, any thoughts?

Secondly (and I sort of alluded to it above), I think having better feed moderation tools would be good. This could be some way for feed owners to officially make someone a feed moderator - then, in-app, they would be able to e.g. edit the list of users or somehow tag posts for removal. Other things could also be useful, like being able to "mute" users from a feed for a certain amount of time. We haven't had any big issues on the astronomy feed _yet_, but I'm sure there will be eventually - and right now, I'm the only person who can moderate the feed easily, which isn't ideal.

Finally, improvements to feed discoverability would be great. At some point, people started getting suggested custom feeds on sign up, which is awesome! I would love to see this continue while people use the app, potentially based on e.g. who they follow. In addition, I think that when the like button for feeds moved to the "about" tab, it made people like feeds much less. This means that older feeds (like the Astronomy one) have much higher like counts than newer feeds, as it's much harder to get feed likes now. Since likes seem to be used to pick which feeds get recommended, I think this probably hurts new feeds from before the like count change, and it makes it harder to launch a new feed and have it be successful. Improvements to the liking system would be good for both existing & future feeds.

And by the way - thank you, as ever, for all that you folks are doing =)

**Reply by bnewbold** [reply]
Thanks for this feedback! A lot of this resonates with the design discussions we have been having, but it is great to hear it coming directly from somebody curating a feed.

**Comment by strideynet** [comment]
re: Community/Interest feeds. Always willing to answer questions/talk about the FurryList stuff - I'll be eagerly keeping an eye on this to see if it'll be a suitable replacement for our hand-crafted feed generator.

**Comment by recalcitrantjim** [comment]
Re: the non public content, why aren't we just encrypting the posts, then giving accounts with permission keys to decrypt them?

**Reply by bnewbold** [reply]
Not a full answer but check out: https://bsky.app/profile/divy.zone/post/3la3f4lmaym25
